pages/add_module.tsx

```diff
@@ -0,0 +1,281 @@
+// Copyright 2022 the Deno authors. All rights reserved. MIT license.
+
+/** @jsx h */
+/** @jsxFrag Fragment */
+import { ComponentChildren, Fragment, h, Head } from "../deps.ts";
+import { CDN_ENDPOINT, getVersionList } from "../util/registry_utils.ts";
+import { Header } from "../components/Header.tsx";
+import { InlineCode } from "../components/InlineCode.tsx";
+import { Footer } from "../components/Footer.tsx";
+
+export default function AddModulePage() {
+  return (
+    <>
+      <Head>
+        <title>Third Party Modules | Deno</title>
+      </Head>
+      <script>
+        {`
+          async function getVersionList(module) {
+            const url = \`${CDN_ENDPOINT}\${module}/meta/versions.json\`;
+            const res = await fetch(url, {
+              headers: {
+                accept: 'application/json',
+              },
+            });
+            if (res.status === 403 || res.status === 404) return null;
+            if (res.status !== 200) {
+              throw Error(
+                \`Got an error (\${res.status}) while getting the version list:\\n\${await res
+            .text()}\`,
+              );
+            }
+            
+            return await res.json();
+          }
+          
+          function updateWebhookUrl() {
+            const name = document.getElementById('modulename').value;
+            const subdir = document.getElementById('subdirectory').value;
+            document.getElementById('webhookURL').innerText = \`https://api.deno.land/webhook/gh/\${name}\${subdir ? ('?subdir=' + encodeURIComponent(subdir)) : ''}\`;
```

Comment:
not for now, but string literals is not the ideal place to be programming... Why isn't fresh's esbuild capabilities not sufficient ?

Reply:
@lucacasonato